stratum/hal/lib/tdi/tdi_sde_table_entry.cc

```diff
@@ -143,9 +147,14 @@ ::util::Status TdiSdeWrapper::DeleteTableEntry(
   const auto flags = ::tdi::Flags(0);
   tdi_status_t status = table->entryDel(
       *real_session->tdi_session_, *dev_tgt, flags, *real_table_key->table_key_);
-  if (!((status == BF_SUCCESS) || (status == BF_OBJECT_NOT_FOUND)))
+
+  if ((status == BF_OBJECT_NOT_FOUND)) {
+     return MAKE_ERROR(::util::error::Code::NOT_FOUND) <<
+             "No matching table entry with " << dump_args();
+  } else if (!(status == BF_SUCCESS)) {
```

Comment:
`(status != BF_SUCCESS)` would be a clearer (and wouldn't require a second set of parentheses)

Reply:
Done